src/main/java/org/assertj/core/internal/Throwables.java

```diff
@@ -133,6 +134,14 @@ public void assertHasRootCause(AssertionInfo info, Throwable actual, Throwable e
       throw failures.failure(info, shouldHaveRootCause(actualRootCause, expectedRootCause));
   }
 
+  public void assertHasRootCauseMessage(AssertionInfo info, Throwable actual, String expectedMessage) {
+    assertNotNull(info, actual);
+    Throwable rootCause = getRootCause(actual);
+    if (null == rootCause) throw failures.failure(info, shouldHaveRootCauseWithMessage(expectedMessage));
+    if (areEqual(rootCause.getMessage(), expectedMessage)) return;
+    throw failures.failure(info, shouldHaveMessage(rootCause, expectedMessage), rootCause.getMessage(), expectedMessage);
```

Comment:
Yes, I agree that we need to improve here. Will take a look.

Reply:
So I have used `ShouldHaveRootCause(Throwable actualCause, String expectedCauseMessage)` to make message more clear.

As for:

> another improvement is to add:

I can do that, but it will require extra logic rewrite as right now `ShouldHaveRootCause(Throwable actualCause, String expectedCauseMessage)` receives root cause which will result not a full stack trace. What do you think?

Reply:
I think it `ShouldHaveRootCause` should take the actual exception as an additional parameter in first position, this means improving the others error message too as a nice side effect of this PR.

it should not be too much work but let me know if you don't have the time (it's ok if that's the case).

Reply:
For this I'll create a follow up PR, if it is ok. I have started to work on it. It touches more than one method so I think it will make more sense to have it separately.